### ISSUE TITLE
[System test runner] Implement result assertion

### ISSUE THREAD

**Issue by mtojek** [elastic/elastic-package#113]
The elastic-package verifies if documents with results are in Elasticsearch, but it doesn't assert any fields of these ones. 

Introduce support for assertions of test results.

**Comment by mtojek** [comment]
I think it would be nice to deliver this feature now :) I will take a look at it.

**Comment by mtojek** [comment]
We're checking type correctness, potential error messages.
The issue for skipped fields has been created: https://github.com/elastic/elastic-package/issues/147

I suppose the next step is smart comparison of expected and collected documents in system tests?

**Comment by ycombinator** [comment]
> I suppose the next step is smart comparison of expected and collected documents in system tests?

Yes, that will complete the scope of this issue.

**Comment by mtojek** [comment]
I think we discussed it some time ago, but it might be a quick win here - 

Add `pattern` property to the string based fields in `fields.yml` files, and perform type assertion based on these fields only. No additional files for storing documents with patterns will be required.

**Comment by ycombinator** [comment]
Yes, I think the quick win makes sense because I think pattern matching is the best we're going to be able to do with system tests anyway (since we're dealing with live data). But, of course, we would support it in pipeline tests as well. And, as you noted, pattern matching will only be applied for fields that define patterns; otherwise they just get the basic existence and type checking you implemented in #143.

And then, in a follow up PR, we can _also_ support exact document matching. The understanding here would be that this only makes sense for pipeline tests.
